src/events.cpp

```diff
@@ -66,52 +55,60 @@ bool context::remove_handler(sdl_handler* ptr)
 	static int depth = 0;
 	++depth;
 
-	//the handler is most likely on the back of the events array,
-	//so look there first, otherwise do a complete search.
+	// The handler is most likely on the back of the events list,
+	// so look there first, otherwise do a complete search.
 	if(handlers.back() == ptr) {
-		delete_handler_index(handlers.size()-1);
+		if(*focused_handler == ptr) {
```

Comment:
You could. If there's only one element, you'd set it to end(), otherwise decrement it.